src/Featurizers/TruncatedSVDFeaturizer.h

```diff
@@ -0,0 +1,631 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "Components/PipelineExecutionEstimatorImpl.h"
+#include "Components/TrainingOnlyEstimatorImpl.h"
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wold-style-cast"
+#   pragma clang diagnostic ignored "-Wsign-conversion"
+#   pragma clang diagnostic ignored "-Wdocumentation-unknown-command"
+#   pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
+#   pragma clang diagnostic ignored "-Wextra-semi-stmt"
+#   pragma clang diagnostic ignored "-Wmissing-noreturn"
+#   pragma clang diagnostic ignored "-Wdocumentation"
+#   pragma clang diagnostic ignored "-Wdouble-promotion"
+#   pragma clang diagnostic ignored "-Wcast-align"
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#   pragma clang diagnostic ignored "-Wshadow"
+#elif (defined _MSC_VER)
+#   pragma warning(push)
+#   pragma warning(disable: 4127)
+#endif
+
+#include "../3rdParty/eigen/Eigen/Dense"
```

Comment:
>../3rdParty/eigen/Eigen/Dense" [](start = 10, length = 30)

Would you rather just state #include "Eigen/Dense" ? For this 1) In the main CMakeLists.txt add interface only target:
    add_library(eigen INTERFACE)
    target_include_directories(eigen INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/3rdParty/eigen)